pkg/endpoint/endpoint.go

```diff
@@ -1281,7 +1188,8 @@ func (e *Endpoint) SetDefaultOpts(opts *option.BoolOptions) {
 	}
 
 	if opts != nil {
-		for k := range EndpointMutableOptionLibrary {
+		epOptLib := option.GetEndpointMutableOptionLibrary()
```

Comment:
This should use `EndpointOptionLibrary` from this package, shouldn't it? (that would also save you this line and make the loop more obvious)